modules/replicator/src/main/java/org/apache/ignite/internal/replicator/listener/ReplicaListener.java

```diff
@@ -31,4 +33,20 @@ public interface ReplicaListener {
      * @return Listener response.
      */
     CompletableFuture<?> invoke(ReplicaRequest request);
+
+    /**
+     * Callback on leader election.
+     *
+     * @param clusterNode Leader node.
+     */
+    default void onLeaderElected(ClusterNode clusterNode) {
```

Comment:
We may have leaderless replication protocols. I'd say that we should use onBecomePrimary instead.

Reply:
fix it